docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md

```diff
@@ -0,0 +1,103 @@
+
+### Extra Information
+
+**How do we press?**
+
+We have 2 options, we can press a batch of interactives using scripts or we can press a single interactive using the frontend admin tool.
+
+For pressing a batch of interactives this is controlled using command line scripts. An example of these scripts is included in the frontend docs. The batch pressing process is separated into a couple of independent steps:
+1. Querying the content API, or via some other means, to determine the list of interactive URLs to press
+2. For every URL make a request to /interactive-librarian/live-presser/{path}
+3. For every successfully pressed article make a request to /interactive-librarian/read-clean-write/{path}
+
+To press a single interactive we can use the frontend admin tool. On the admin tool, select the new option ‘Press an interactive’, enter the full URL for the interactive, click ‘Press’ and wait for the response. If there’s an error in the response it’ll need to be reported to the dotcom team.
+
+**How do we know a page is pressed?**
+
+Each pressed article will have additional copy appended to the document. At the very bottom of the page we should see the copy ‘This article was archived on <date>. Some elements may be out of date.’
+
+**How do we decide to serve pressed content to readers?**
+
+We have an intermediate solution for determining if a page is pressed: we retain a list of article URLs in config. Only if the article being requested is included in this list do we serve the pressed content.
+
+In the long term we’d like to mark pressed articles with a tracking tag (tracking/dcroptout). Articles with this tag will be served as pressed versions. We can’t adopt this solution until Ed Tools have completed a spike to understand how we can achieve this.
+
+**How can we view a pressed page?**
+To view a pressed page there are a couple of options:
+- Get the document from S3 directly (aws-frontend-archive).
+- Intermediate solution: add the interactive path to the frontend config (https://github.com/guardian/frontend/blob/dlawes/serve-pressed-interactives/common/app/services/dotcomrendering/PressedInteractives.scala#L11).
+- Long-term solution: add tag tracking/dcroptout to article.
+
+**Can we opt-out of pressing and render via frontend or DCR?**
+
+Yes! In the short term, if we want to opt-out of serving pressed content then we omit or remove the interactive path in the relevant frontend config. In the long term, if we want to opt-out of serving the pressed content we will remove, or we will not apply, the chosen tracking tag (tracking/dcroptout).
+
+Note that, after some agreed period of time, I believe our intention would be to remove the rendering code from the frontend codebase, after which point articles could only be rendered via DCR.
```

Comment:
I would firm this bit up: `I believe our intention` -> `The plan is...`.

Reply:
Also, how can we ensure this happens? A Trello card might get lost. Perhaps a Dotcom calendar event linked to a Trello card describing the work (and including a link to this doc too)?

And thoughts on a reasonable timeframe? Obviously need to discuss with Rob+Visuals, but 3 months might be a good balance between enough time to provide safety, while also realising the benefits of deleting the code soon as well.

Reply:
Great suggestion - I'll create a trello card and a calendar event, will link both to this doc